gradio/cli/commands/components/build.py

```diff
@@ -25,6 +30,29 @@ def _build(
         )
         if build_frontend:
             live.update(":art: Building frontend")
+            component_directory = path.resolve()
+
+            node = shutil.which("node")
+            if not node:
+                raise ValueError("node must be installed in order to run dev mode.")
+
+            node_cmds = [
+                node,
+                gradio_node_path,
+                "--component-directory",
+                component_directory,
+                "--root",
+                gradio_template_path,
+                "--mode",
+                "build",
+            ]
+
+            pipe = subprocess.run(node_cmds, capture_output=True, text=True)
+            if pipe.returncode != 0:
```

Comment:
If the front-end build fails we should terminate I think

Reply:
I just returned out of the function, is that the correct approach?

Reply:
Yea that looks good!